Sources/LeafKit/LeafSerialize/LeafSerializer.swift

```diff
@@ -85,26 +77,11 @@ internal struct LeafSerializer {
             body: tag.body,
             userInfo: self.userInfo
         )
-        let rendered = try self.tags[tag.name]?.render(sub)
-            ?? .init(.null)
-        serialize(rendered)
+        let leafData = try self.tags[tag.name]?.render(sub) ?? LeafData.trueNil
+        try? leafData.serialize(buffer: &self.buffer)
```

Comment:
Slightly more efficient - leafData will now serialize themselves into a provided buffer instead of having to branch into yet another generic "serialize" with unclear parameter labels.